### ISSUE TITLE
research: test whether an oracle-grounded synthetic SWE pipeline adds capability without degrading model performance

### ISSUE THREAD

**Issue by rmems** [rmems/synthetic-factory#153]
## Planning / research attribution

**GPT-5.6-Sol (OpenAI / ChatGPT), 2026-08-28**, with Raul Montoya Cardenas directing the Synthetic Factory research question and downstream post-training comparison.

## Research question

The goal of the next software-engineering synthetic-data program is **not** to rank GPT-5.6-Sol, Meta Muse Spark 1.2, and Claude Fable 5 against each other as data generators.

The question is:

> Can a provenance-rich, oracle-grounded synthetic-data pipeline produce training data that improves useful software-engineering behavior without degrading the starting model, and does it add incremental value beyond strong external public SWE corpora and provenance-rich real engineering trajectories from Operation Prometheus?

Model-specific roles inside the factory are implementation details / hypotheses. The research result is judged downstream by model behavior.

## Required downstream comparison

The authoritative downstream experiment is `rmems/agoge-forger#103` / rmems/agoge-forger#103.

Using one exact pinned base-model revision and one frozen evaluation contract, compare:

```text
A — BASE
exact starting checkpoint; no fine-tuning

B — SYNTHETIC FACTORY ONLY
curated Synthetic Factory SWE release

C — EXTERNAL PUBLIC SWE ONLY
approved third-party curated SWE / coding-agent corpus mixture

D — PROMETHEUS ONLY
curated rmems/operation-prometheus SFT materialization

E — PUBLIC + PROMETHEUS
strong non-Synthetic-Factory baseline

F — SYNTHETIC + PUBLIC + PROMETHEUS
same total training budget as E for the primary composition test
```

For B/C/D/E/F, hold constant as far as practical:

* accepted training-token budget;
* optimizer / LR / epochs or effective update budget;
* sequence length;
* LoRA/QLoRA method and target modules;
* seed policy;
* checkpoint-selection policy;
* inference/evaluation settings.

Freeze source membership and first-pass mixture ratios before measured results are inspected.

## Primary hypotheses

H1 — Synthetic Factory-only SFT can improve targeted engineering behavior over the base model without exceeding a predeclared degradation margin on protected regression metrics.

H2 — External curated public SWE corpora provide a strong established agent-training baseline.

H3 — Operation Prometheus contributes provenance-rich real engineering signal beyond public SWE corpora.

H4 — Adding Synthetic Factory data to the strongest public+Prometheus baseline can improve transfer, or at minimum remain within a predeclared non-inferiority margin.

Negative, null, or mixed results are valid outcomes.

## Flagship comparison

The most important Synthetic Factory result is:

```text
F - E
```

Does Synthetic Factory add useful signal when the model already has strong external public SWE data plus curated real engineering history?

If F regresses relative to E beyond the frozen margin, that is evidence against the current synthetic-data composition/curation policy and must be reported honestly.

## Source taxonomy

Do not classify every third-party Hugging Face trajectory dataset as `real`.

Preserve at least:

```text
SYNTHETIC_FACTORY
controlled generated tasks / trajectories produced and curated by rmems/synthetic-factory

EXTERNAL_PUBLIC_SWE
third-party curated SWE / coding-agent corpora
may itself contain synthetic trajectories over real or synthetic tasks

PROMETHEUS_REAL
rmems/operation-prometheus
real public GitHub engineering-history trajectories with explicit lineage/provenance
```

The exact provenance class must survive into manifests and reports.

## External public SWE inputs

Candidate public sources are selected and admitted downstream by rmems/agoge-forger#103 under a source-admission contract covering license, schema, provenance, hidden-reasoning policy, deduplication, and held-out contamination.

The candidate pool currently includes corpora such as:

* `nvidia/Open-SWE-Traces`
* `togethercomputer/CoderForge-Preview`
* `open-thoughts/OpenThoughts-Agent-SFT-100K`
* `nvidia/SWE-Hero-openhands-trajectories`
* `SWE-bench/SWE-smith-trajectories`
* `nebius/SWE-rebench-openhands-trajectories`
* `R2E-Gym/R2EGym-SFT-Trajectories`
* `AmanPriyanshu/tool-reasoning-sft-CODING-MEnvData-SWE-Trajectory-data-cleaned-rectified`
* other reviewed public sources that pass the same admission gate

This list is a candidate pool, not automatic inclusion.

## No-degradation contract

Before the measured baseline is inspected, downstream Agoge execution must freeze:

* primary task-success metric(s);
* protected regression metric(s);
* non-inferiority / degradation margin(s);
* invalid/inconclusive handling;
* evaluation sample membership;
* seed/replication policy.

Protected metrics must include at least one capability slice independent of Synthetic Factory, Operation Prometheus, and the admitted external public SWE training sources.

Training loss is not evidence of improvement.

## Synthetic pipeline definition

A synthetic record is admissible only when the pipeline can distinguish:

```text
task author
solver / execution agent
reviewer / critic
oracle
curation decision
release provenance
```

Generators may propose tasks, patches, reviews, or counterexamples. They do not self-certify correctness.

Accepted labels must be grounded by executable or otherwise validated evidence such as:

* compiler/test results;
* validated hidden/property/metamorphic tests;
* mutation testing where useful;
* trusted reference behavior;
* deterministic fixtures;
* explicit human adjudication for irreducibly semantic cases.

## Required source separation

Synthetic Factory, external public SWE corpora, and Operation Prometheus remain different provenance classes.

* `synthetic-factory` owns generated task/data production, oracle validation, curation, and release.
* external public SWE corpora are third-party inputs normalized/admitted by Agoge before the comparative experiment.
* `operation-prometheus` owns real public engineering-history trajectories and derived real-world SFT/eval material.
* `agoge-forger` owns cross-source normalization for training, matched-budget post-training, and A–F evaluation.

Do not fabricate GitHub provenance for synthetic records and do not rewrite Prometheus records as synthetic examples.

## Hard gates before material generation

Corpus-scale GPT/Muse/Fable generation is blocked until the current factory control plane is ready:

- [ ] #1 — strict curation/promotion of the existing corpus
- [ ] rmems/synthetic-factory#23 / GitHub #23 — composed fail-closed `training_ready` export
- [ ] rmems/synthetic-factory#26 / GitHub #26 — hidden/internal reasoning stripped from training views
- [ ] rmems/synthetic-factory#32 / GitHub #32 — registry-backed multi-generator authority
- [ ] GitHub #13 / [RM-645](https://linear.app/rpd-34/issue/RM-645/sf-0qz-token-efficient-windows-early-stop-on-coverage-convergence) — context hygiene / generation-efficiency contract is ready for another material run

Fixture-scale generation for schemas, validators, and oracle tests remains allowed.

## Existing real-world anchors

Downstream comparison should consume, when stable:

* rmems/operation-prometheus#54 / `rmems/operation-prometheus#54` — issue-to-patch SFT materialization
* rmems/operation-prometheus#55 / `rmems/operation-prometheus#55` — time-forward, lineage-safe held-out engineering tasks

The held-out Prometheus pool must be frozen before deriving synthetic variants from real engineering patterns. No evaluation lineage may seed Synthetic Factory training data or survive as a duplicate/near-duplicate in admitted external public training data.

## Downstream Agoge anchors

Reuse rather than duplicate:

* rmems/agoge-forger#76 / `rmems/agoge-forger#76` — first-class SFT baseline
* rmems/agoge-forger#99 / #99 — immutable split contract / leakage guards
* rmems/agoge-forger#100 / #100 — base-vs-SFT evaluation harness
* rmems/agoge-forger#101 / #101 — first controlled SFT pilot
* rmems/agoge-forger#103 / #103 — A–F Synthetic Factory vs public SWE vs Prometheus matched-budget ablation

## Synthetic release success metrics

Factory-side quality should report at least:

* oracle-valid records;
* curation survival;
* duplicate-adjusted coverage;
* human acceptance sample;
* invalid/impossible task rate;
* evidence completeness;
* accepted records per model budget / wall time.

These are process metrics, not proof of model benefit. Downstream A–F evaluation is the final utility test.

## Relationships

* **Blocked by:** #1, #13, #23, #26, #32
* **Sibling research program: **#76 (oracle-grounded neuromorphic datasets)
* **Real-data dependencies: **`operation-prometheus#54`, `operation-prometheus#55`
* **Training/eval dependencies: **`agoge-forger#76`, `#99`, `#100`, `#101`, `#103`

## Acceptance criteria

- [ ] One synthetic SWE release can pass schema → oracle validation → curation → fail-closed export.
- [ ] Synthetic Factory, external public SWE, and Prometheus provenance remain structurally distinct.
- [ ] The downstream A–F experiment is frozen before measured outcomes are inspected.
- [ ] B/C/D/E/F use matched accepted-token budgets for the primary comparison.
- [ ] Base and all trained arms use the same held-out evaluation contract.
- [ ] `F - E` is reported explicitly as the incremental Synthetic Factory composition effect over the strongest public+real baseline.
- [ ] Improvement, degradation, mixed, null, and inconclusive outcomes can all be represented honestly.
- [ ] No decision to scale a synthetic family is based only on repository count or raw-record volume.

## Tracking

GitHub is source of truth. rmems/synthetic-factory#153 mirrors `rmems/synthetic-factory#153` in the existing Synthetic Data Factory project. Preserve current project/milestone/relationship metadata rather than creating another twin.

**Comment by rmems** [comment]
## Downstream / child graph locked

Created the approved implementation children:

- #154 — actor-role provenance
- #155 — single verified SWE synthetic release candidate
- #156 — Fable 5 repo-pack/task-author stage
- #157 — GPT-5.6-Sol execution/solver stage
- #158 — Meta Muse Spark 1.2 adversarial review/oracle-hardening stage

The downstream model-level test is now **rmems/agoge-forger#103**: base vs synthetic-only vs Operation Prometheus real-only vs matched synthetic+real SFT. The real training/evaluation anchors remain **rmems/operation-prometheus#54** and **#55**.

Research interpretation remains pipeline-first: model-specific roles are production stages, not the primary comparison. Synthetic utility is earned only through downstream transfer/no-degradation evidence.

**Research/design attribution:** GPT-5.6-Sol (OpenAI / ChatGPT), with Raul Montoya Cardenas directing the research question and approval. 2026-08-28.

**Comment by rmems** [comment]
**DELETE** — recommendation from **01 Synthetic Factory Debt Closer** (Tech Debt Mill).

Debt-mode freeze: no new train/gen, VSET–SWE campaign, or neuromorphic dataset-family work. Close unless Raul explicitly reopens for a named mill.

**Comment by rmems** [comment]
**Status update 2026-09-04**

Not started. Under #161 the ablation may only consume training-candidate data, so it now depends on the procedural/simulator lane (#169, #170) or a DeepSeek/Nemotron lane after #163; the frontier corpora are research comparators only. No blocker beyond those dependencies.

— Claude Fable 5.1 (Claude Code)

**Comment by rmems** [comment]
Closed per Tech Debt Mill DELETE triage (debt mode: not planned / frozen train-experiment-research). Reopen only with a named brief.
